### PR TITLE
Add share feature

### PR DESCRIPTION
**Pull request by nelsontky**
Coupon is saved as an image file like below:
Auntie Azah's Nasi Lemak

Tests are not added yet, thinking of using SHA256 checksums or something to test. However, getting the `FileChooser` dialog to work on the Travis CI may be an issue so KIV...

Annnnddddd I added something in `build.gradle` so it must be declared?

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,6 +57,10 @@ dependencies {
     implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'win'
     implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'mac'
     implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'linux'
+    implementation group: 'org.openjfx', name: 'javafx-swing', version: javaFxVersion, classifier: 'win'
```
Comment (nelsontky) [inline build.gradle]: I added `javafx-swing` in in `build.gradle` cuz it was needed to convert an `FxImage` to `png`.

File: build.gradle [hunk 2]
```diff
@@ -57,6 +57,10 @@ dependencies {
...
+    implementation group: 'org.openjfx', name: 'javafx-swing', version: javaFxVersion, classifier: 'mac'
+    implementation group: 'org.openjfx', name: 'javafx-swing', version: javaFxVersion, classifier: 'linux'
```
Comment (nelsontky) [inline build.gradle:62]: I added `javafx-swing` in `build.gradle` cuz it was needed to convert an `FxImage` to `png`.

Comment (nelsontky) [reply build.gradle:62]: I've posted a request at https://github.com/nus-cs2103-AY1920S2/forum/issues/90

**Review by ChesterSim — APPROVED** [review]
LGTM!

**Pull request merged**